domains/transfer-agreements/TransferAgreements.API/API/Program.cs

```diff
@@ -33,6 +37,40 @@
 builder.Services.AddOptions<DatabaseOptions>().BindConfiguration(DatabaseOptions.Prefix).ValidateDataAnnotations().ValidateOnStart();
 builder.Services.AddDbContext<ApplicationDbContext>((sp, options) => options.UseNpgsql(sp.GetRequiredService<IOptions<DatabaseOptions>>().Value.ToConnectionString()));
 
+Audit.Core.Configuration.Setup()
+    .UseEntityFramework(ef => ef
+        .AuditTypeExplicitMapper(config => config
+            .Map<TransferAgreement, TransferAgreementHistoryEntry>((evt, eventEntry, historyEntity) =>
+            {
+                if (eventEntry.Entity != null && eventEntry.Entity.GetType() == typeof(TransferAgreementHistoryEntry))
+                {
+                    return false;
+                }
```

Comment:
Why is this check needed? The Map method states that it is a mapping for TransferAgreement -> TransferAgreementHistoryEntry, so I do not understand why this check is there.

Reply:
Because, if it is not there, Audit.net will try auditing the history entry table. Will research further, for a better solution

Reply:
It is to disable auditing the history entry table, when seeding the database in the test environment. The SeedData method, Might be implemented incorrectly. Will research further.

Reply:
SeedData method, in factory, has been refactored, thus we do not need this if statement anymore. It has been removed.